In [3]:
# importacion general de librerias y de visualizacion (matplotlib y seaborn)
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

plt.style.use('default') # haciendo los graficos un poco mas bonitos en matplotlib
#plt.rcParams['figure.figsize'] = (20, 10)

sns.set(style="whitegrid") # seteando tipo de grid en seaborn

In [8]:
types= {
    "event": "category", 
    "model": "category", 
    "condition": "category", 
    "color": "category", 
    "storage": "category",
}
parse_dates = ['timestamp']
df_events = pd.read_csv('./fiuba-trocafone-tp1-final-set/events.csv', dtype=types, low_memory=False, parse_dates=parse_dates)

In [31]:
df_events['year'] = df_events['timestamp'].dt.year
df_events['month'] = df_events['timestamp'].dt.month
df_events['day'] = df_events['timestamp'].dt.day
df_events['time'] = df_events['timestamp'].dt.time

df_events

timestamp              event    person  \
0       2018-05-31 23:38:05    ad campaign hit  0004b0a2   
1       2018-05-31 23:38:05       visited site  0004b0a2   
2       2018-05-31 23:38:09     viewed product  0004b0a2   
3       2018-05-31 23:38:40           checkout  0004b0a2   
4       2018-05-29 13:29:25     viewed product  0006a21a   
5       2018-05-29 13:29:26       visited site  0006a21a   
6       2018-05-29 13:29:27    ad campaign hit  0006a21a   
7       2018-05-29 13:29:35           checkout  0006a21a   
8       2018-04-09 20:12:31    generic listing  000a54b2   
9       2018-04-09 20:12:31       visited site  000a54b2   
10      2018-04-09 20:12:31  search engine hit  000a54b2   
11      2018-04-09 20:12:31    ad campaign hit  000a54b2   
12      2018-04-09 20:12:53      brand listing  000a54b2   
13      2018-04-09 20:13:14     viewed product  000a54b2   
14      2018-04-09 20:13:20      brand listing  000a54b2   
15      2018-05-24 11:21:07    ad campaign hit  000a54b2   
16      2018-05-24 11:21:07       visited site  000a54b2   
17      2018-05-24 11:21:07  search engine hit  000a54b2   
18      2018-05-24 11:21:08    generic listing  000a54b2   
19      2018-05-24 11:26:34    generic listing  000a54b2   
20      2018-05-24 11:26:53      brand listing  000a54b2   
21      2018-05-24 11:26:56    generic listing  000a54b2   
22      2018-05-24 11:27:47     viewed product  000a54b2   
23      2018-05-24 11:27:50    generic listing  000a54b2   
24      2018-05-24 11:28:10      brand listing  000a54b2   
25      2018-05-24 11:28:36      brand listing  000a54b2   
26      2018-05-24 11:28:59     viewed product  000a54b2   
27      2018-05-24 11:29:06     viewed product  000a54b2   
28      2018-05-24 11:29:14     viewed product  000a54b2   
29      2018-05-24 11:29:16     viewed product  000a54b2   
...                     ...                ...       ...   
1011258 2018-06-09 13:40:05       visited site  ffffa8d1   
1011259 2018-06-09 13:40:05  search engine hit  ffffa8d1   
1011260 2018-06-09 13:40:05    generic listing  ffffa8d1   
1011261 2018-06-09 13:42:05  searched products  ffffa8d1   
1011262 2018-06-09 13:43:43     viewed product  ffffa8d1   
1011263 2018-06-09 13:43:52     viewed product  ffffa8d1   
1011264 2018-06-09 15:33:23     viewed product  ffffa8d1   
1011265 2018-06-09 15:33:30     viewed product  ffffa8d1   
1011266 2018-06-09 15:34:46     viewed product  ffffa8d1   
1011267 2018-06-09 15:35:00     viewed product  ffffa8d1   
1011268 2018-06-09 15:35:11     viewed product  ffffa8d1   
1011269 2018-06-09 15:35:13     viewed product  ffffa8d1   
1011270 2018-06-09 15:35:18     viewed product  ffffa8d1   
1011271 2018-06-09 15:39:05     viewed product  ffffa8d1   
1011272 2018-06-09 15:39:07     viewed product  ffffa8d1   
1011273 2018-06-09 15:39:56       visited site  ffffa8d1   
1011274 2018-06-09 15:41:29    generic listing  ffffa8d1   
1011275 2018-06-09 15:43:29     viewed product  ffffa8d1   
1011276 2018-06-11 12:25:40  search engine hit  ffffa8d1   
1011277 2018-06-11 12:25:40       visited site  ffffa8d1   
1011278 2018-06-11 12:25:40    ad campaign hit  ffffa8d1   
1011279 2018-06-11 12:25:41    generic listing  ffffa8d1   
1011280 2018-06-11 12:28:47  searched products  ffffa8d1   
1011281 2018-06-11 12:29:11     viewed product  ffffa8d1   
1011282 2018-06-11 12:29:14     viewed product  ffffa8d1   
1011283 2018-06-11 12:31:49           checkout  ffffa8d1   
1011284 2018-06-04 23:18:03       visited site  ffffac8a   
1011285 2018-06-04 23:18:03    ad campaign hit  ffffac8a   
1011286 2018-06-04 23:18:03     viewed product  ffffac8a   
1011287 2018-06-04 23:18:10           checkout  ffffac8a   

                                     url      sku                   model  \
0              /comprar/iphone/iphone-5s      NaN                     NaN   
1                                    NaN      NaN                     NaN   
2                                    NaN   2694.0               iP

In [40]:
df_events.columns

Index(['timestamp', 'event', 'person', 'url', 'sku', 'model', 'condition',
       'storage', 'color', 'skus', 'search_term', 'staticpage',
       'campaign_source', 'search_engine', 'channel', 'new_vs_returning',
       'city', 'region', 'country', 'device_type', 'screen_resolution',
       'operating_system_version', 'browser_version'],
      dtype='object')

timestamp              event    person  \
0       2018-05-31 23:38:05    ad campaign hit  0004b0a2   
1       2018-05-31 23:38:05       visited site  0004b0a2   
2       2018-05-31 23:38:09     viewed product  0004b0a2   
3       2018-05-31 23:38:40           checkout  0004b0a2   
4       2018-05-29 13:29:25     viewed product  0006a21a   
5       2018-05-29 13:29:26       visited site  0006a21a   
6       2018-05-29 13:29:27    ad campaign hit  0006a21a   
7       2018-05-29 13:29:35           checkout  0006a21a   
8       2018-04-09 20:12:31    generic listing  000a54b2   
9       2018-04-09 20:12:31       visited site  000a54b2   
10      2018-04-09 20:12:31  search engine hit  000a54b2   
11      2018-04-09 20:12:31    ad campaign hit  000a54b2   
12      2018-04-09 20:12:53      brand listing  000a54b2   
13      2018-04-09 20:13:14     viewed product  000a54b2   
14      2018-04-09 20:13:20      brand listing  000a54b2   
15      2018-05-24 11:21:07    ad campaign hit  000a54b2   
16      2018-05-24 11:21:07       visited site  000a54b2   
17      2018-05-24 11:21:07  search engine hit  000a54b2   
18      2018-05-24 11:21:08    generic listing  000a54b2   
19      2018-05-24 11:26:34    generic listing  000a54b2   
20      2018-05-24 11:26:53      brand listing  000a54b2   
21      2018-05-24 11:26:56    generic listing  000a54b2   
22      2018-05-24 11:27:47     viewed product  000a54b2   
23      2018-05-24 11:27:50    generic listing  000a54b2   
24      2018-05-24 11:28:10      brand listing  000a54b2   
25      2018-05-24 11:28:36      brand listing  000a54b2   
26      2018-05-24 11:28:59     viewed product  000a54b2   
27      2018-05-24 11:29:06     viewed product  000a54b2   
28      2018-05-24 11:29:14     viewed product  000a54b2   
29      2018-05-24 11:29:16     viewed product  000a54b2   
...                     ...                ...       ...   
1011258 2018-06-09 13:40:05       visited site  ffffa8d1   
1011259 2018-06-09 13:40:05  search engine hit  ffffa8d1   
1011260 2018-06-09 13:40:05    generic listing  ffffa8d1   
1011261 2018-06-09 13:42:05  searched products  ffffa8d1   
1011262 2018-06-09 13:43:43     viewed product  ffffa8d1   
1011263 2018-06-09 13:43:52     viewed product  ffffa8d1   
1011264 2018-06-09 15:33:23     viewed product  ffffa8d1   
1011265 2018-06-09 15:33:30     viewed product  ffffa8d1   
1011266 2018-06-09 15:34:46     viewed product  ffffa8d1   
1011267 2018-06-09 15:35:00     viewed product  ffffa8d1   
1011268 2018-06-09 15:35:11     viewed product  ffffa8d1   
1011269 2018-06-09 15:35:13     viewed product  ffffa8d1   
1011270 2018-06-09 15:35:18     viewed product  ffffa8d1   
1011271 2018-06-09 15:39:05     viewed product  ffffa8d1   
1011272 2018-06-09 15:39:07     viewed product  ffffa8d1   
1011273 2018-06-09 15:39:56       visited site  ffffa8d1   
1011274 2018-06-09 15:41:29    generic listing  ffffa8d1   
1011275 2018-06-09 15:43:29     viewed product  ffffa8d1   
1011276 2018-06-11 12:25:40  search engine hit  ffffa8d1   
1011277 2018-06-11 12:25:40       visited site  ffffa8d1   
1011278 2018-06-11 12:25:40    ad campaign hit  ffffa8d1   
1011279 2018-06-11 12:25:41    generic listing  ffffa8d1   
1011280 2018-06-11 12:28:47  searched products  ffffa8d1   
1011281 2018-06-11 12:29:11     viewed product  ffffa8d1   
1011282 2018-06-11 12:29:14     viewed product  ffffa8d1   
1011283 2018-06-11 12:31:49           checkout  ffffa8d1   
1011284 2018-06-04 23:18:03       visited site  ffffac8a   
1011285 2018-06-04 23:18:03    ad campaign hit  ffffac8a   
1011286 2018-06-04 23:18:03     viewed product  ffffac8a   
1011287 2018-06-04 23:18:10           checkout  ffffac8a   

                                     url      sku                   model  \
0              /comprar/iphone/iphone-5s      NaN                     NaN   
1                                    NaN      NaN                     NaN   
2                                    NaN   2694.0               iP

In [9]:
df_events.dtypes

timestamp                   datetime64[ns]
event                             category
person                              object
url                                 object
sku                                 object
model                             category
condition                         category
storage                           category
color                             category
skus                                object
search_term                         object
staticpage                          object
campaign_source                     object
search_engine                       object
channel                             object
new_vs_returning                    object
city                                object
region                              object
country                             object
device_type                         object
screen_resolution                   object
operating_system_version            object
browser_version                     object
dtype: obje

In [30]:
df_events['event'].describe()

count            1011288
unique                11
top       viewed product
freq              528931
Name: event, dtype: object

#### Como hay solo 11 valores distintos lo tomo como un categorico

In [28]:
df_events['model'].describe()

count       564284
unique         202
top       iPhone 6
freq         50916
Name: model, dtype: object

#### Como hay solo 202 valores distintos, pero se contaron 564284 valores, lo tomo como un categorico

In [32]:
df_events.describe()

year         month           day
count  1011288.0  1.011288e+06  1.011288e+06
mean      2018.0  4.838374e+00  1.520886e+01
std          0.0  1.113527e+00  8.772023e+00
min       2018.0  1.000000e+00  1.000000e+00
25%       2018.0  4.000000e+00  8.000000e+00
50%       2018.0  5.000000e+00  1.500000e+01
75%       2018.0  6.000000e+00  2.300000e+01
max       2018.0  6.000000e+00  3.100000e+01

In [38]:
df_events[df_events['storage'].isnull() == False]

timestamp           event    person  url      sku  \
2       2018-05-31 23:38:09  viewed product  0004b0a2  NaN   2694.0   
3       2018-05-31 23:38:40        checkout  0004b0a2  NaN   2694.0   
4       2018-05-29 13:29:25  viewed product  0006a21a  NaN  15338.0   
7       2018-05-29 13:29:35        checkout  0006a21a  NaN  15338.0   
13      2018-04-09 20:13:14  viewed product  000a54b2  NaN  12661.0   
22      2018-05-24 11:27:47  viewed product  000a54b2  NaN  10254.0   
26      2018-05-24 11:28:59  viewed product  000a54b2  NaN   6581.0   
27      2018-05-24 11:29:06  viewed product  000a54b2  NaN   6579.0   
28      2018-05-24 11:29:14  viewed product  000a54b2  NaN   6580.0   
29      2018-05-24 11:29:16  viewed product  000a54b2  NaN   6581.0   
30      2018-05-24 11:29:19  viewed product  000a54b2  NaN   6580.0   
31      2018-05-24 11:29:23  viewed product  000a54b2  NaN   8965.0   
32      2018-05-24 11:30:07  viewed product  000a54b2  NaN   6580.0   
33      2018-05-24 11:30:09  viewed product  000a54b2  NaN   6581.0   
34      2018-05-24 11:30:11  viewed product  000a54b2  NaN   6580.0   
35      2018-05-24 11:30:15  viewed product  000a54b2  NaN   6579.0   
36      2018-05-24 11:30:16  viewed product  000a54b2  NaN   6581.0   
37      2018-05-24 11:30:18  viewed product  000a54b2  NaN   6580.0   
38      2018-05-24 11:30:22  viewed product  000a54b2  NaN   6874.0   
39      2018-05-24 11:30:24  viewed product  000a54b2  NaN   6930.0   
40      2018-05-24 11:30:26  viewed product  000a54b2  NaN   6902.0   
41      2018-05-24 11:30:29  viewed product  000a54b2  NaN   6916.0   
43      2018-05-24 11:33:23  viewed product  000a54b2  NaN  12660.0   
44      2018-05-24 11:34:32        checkout  000a54b2  NaN  12660.0   
48      2018-04-06 05:12:26  viewed product  00184bf9  NaN    102.0   
49      2018-04-06 05:12:27  viewed product  00184bf9  NaN   2716.0   
50      2018-04-06 05:12:30  viewed product  00184bf9  NaN   2751.0   
51      2018-04-06 05:12:32  viewed product  00184bf9  NaN   2709.0   
52      2018-04-06 05:12:33  viewed product  00184bf9  NaN     87.0   
53      2018-04-06 05:12:35  viewed product  00184bf9  NaN    101.0   
...                     ...             ...       ...  ...      ...   
1011234 2018-05-25 18:20:05        checkout  fff60213  NaN   6371.0   
1011237 2018-05-28 18:19:33  viewed product  fff99b85  NaN   2835.0   
1011238 2018-05-28 18:20:01  viewed product  fff99b85  NaN   2837.0   
1011239 2018-05-28 18:20:25        checkout  fff99b85  NaN   2837.0   
1011244 2018-05-15 18:33:08  viewed product  ffff8106  NaN   6649.0   
1011245 2018-05-15 18:33:53  viewed product  ffff8106  NaN   6663.0   
1011246 2018-05-15 18:34:01  viewed product  ffff8106  NaN   6649.0   
1011247 2018-05-15 18:34:21        checkout  ffff8106  NaN   6649.0   
1011252 2018-05-18 00:30:23  viewed product  ffffa8d1  NaN   8554.0   
1011253 2018-05-18 00:30:33  viewed product  ffffa8d1  NaN   8553.0   
1011254 2018-05-18 00:30:37  viewed product  ffffa8d1  NaN   8554.0   
1011255 2018-05-18 00:30:41  viewed product  ffffa8d1  NaN  11709.0   
1011256 2018-05-18 00:30:47  viewed product  ffffa8d1  NaN   8540.0   
1011262 2018-06-09 13:43:43  viewed product  ffffa8d1  NaN   8513.0   
1011263 2018-06-09 13:43:52  viewed product  ffffa8d1  NaN   8512.0   
1011264 2018-06-09 15:33:23  viewed product  ffffa8d1  NaN   8511.0   
1011265 2018-06-09 15:33:30  viewed product  ffffa8d1  NaN   8512.0   
1011266 2018-06-09 15:34:46  viewed product  ffffa8d1  NaN   8526.0   
1011267 2018-06-09 15:35:00  viewed product  ffffa8d1  NaN   9231.0   
1011268 2018-06-09 15:35:11  viewed product  ffffa8d1  NaN   9230.0   
1011269 2018-06-09 15:35:13  viewed product  ffffa8d1  NaN   9231.0   
1011270 2018-06-09 15:35:18  viewed product  ffffa8d1  NaN   8512.0   
1011271 2018-06-09 15:39:05  viewed product  ffffa8d1  NaN   8513.0   
1011272 2018-06-09 15:39:07  viewed product  ffffa8d1  NaN   8512.0   
1011275 2018-06-09 15:43:29  viewed produc